pkg/jx/cmd/opts/common.go

```diff
@@ -219,8 +219,9 @@ func (o *CommonOptions) KubeClient() (kubernetes.Interface, error) {
 			return nil, err
 		}
 		o.kubeClient = kubeClient
-		o.currentNamespace = currentNs
-
+		if o.currentNamespace == "" {
+			o.currentNamespace = currentNs
```

Comment:
So this is the actual "fix", right? The rest is improving on error reporting, right?

Reply:
right